### PR TITLE
Add SessionLog initialization option to not lock session log files

### LINKED ISSUE #518: OpenTap build improvement: log files are locked
Log files are locked (blocking e.g. git clean -dfx) - OpenTAP tasks should release these text files

This is an issue because dotnet reuses processes and tasks between builds, and the msbuild tasks shipped by OpenTAP do not release their held resources after a task finishes -- this is usually done when the process exits, which would usually be after a CLI action ends, but this is not how OpenTAP is used in build tasks. We should add setup and cleanup code to the tasks.

### PR DESCRIPTION
**Pull request by alnlarsen**
This fixes an issue that prevented `git clean -dfx` from working because OpenTAP keeps log files open.

Closes #518

### CODE REVIEW

File: Engine/SessionLogs.cs [hunk 1]
```diff
@@ -73,6 +74,17 @@ public static void Initialize()
...
         {
+            // We can't just add this as a parameter with a default value because
+            // it isn't backwards compatible with plugins compiled against older versions.
+            Initialize(tempLogFileName, canDeleteLog);
+        }
+        
+        /// <summary>
+        /// Initializes the logging. 
+        /// </summary>
+        public static void Initialize(string tempLogFileName, bool canDelete)
```
Comment (rmadsen-ks) [inline Engine/SessionLogs.cs:85]: Can we add to the xml comment what canDelete does and possibly improve the name maybe 'noWriteLock' or noExclusiveWriteLock?

File: Engine/SessionLogs.cs [hunk 2]
```diff
@@ -73,6 +74,17 @@ public static void Initialize()
         /// </summary>
         public static void Initialize(string tempLogFileName)
         {
+            // We can't just add this as a parameter with a default value because
+            // it isn't backwards compatible with plugins compiled against older versions.
+            Initialize(tempLogFileName, canDeleteLog);
```
Comment (rmadsen-ks) [inline Engine/SessionLogs.cs:79]: I dont understand this comment. Why is it not compatible with older plugins?

**Review by rmadsen-ks — CHANGES_REQUESTED** [review]

File: Engine/SessionLogs.cs (hunk 2 above)
Comment (alnlarsen) [reply Engine/SessionLogs.cs:79]: Because on the IL level that would mean removing this overload and adding a method with a different signature

**Comment by alnlarsen** [comment]
I agree with this feedback. Please have another look!

**Review by rmadsen-ks — CHANGES_REQUESTED** [review]
It seems the build logs does not get flushed when the build ends.

image

**Comment by alnlarsen** [comment]
> It seems the build logs does not get flushed when the build ends.
> 
> image

That's a good point! I have added a flush after the build task ends.

**Review by rmadsen-ks — APPROVED** [review]
Seems to work perfectly now!

**Comment by rmadsen-ks** [comment]
Note, now the log files are separated for each msbuild process:
``` 
ls bin/Debug/SessionLogs/
MSBuild.23828 2022-05-12 13-36-31 (1).txt  MSBuild.23828 2022-05-12 13-36-31.txt  MSBuild.26736 2022-05-12 13-36-31.txt
```

**Review by AsgerIversen — APPROVED** [review]
Thanks @alnlarsen. Looks good now

**Pull request merged**